### ISSUE TITLE
Bug: Markdown shortcuts for code are triggered ahead of time

### ISSUE THREAD

**Issue by moy2010** [facebook/lexical#3903]
The markdown shortcuts for code formating (``) and code block (```) get triggered "ahead-of-time", sort of. I don't recall observing this behaviour in earlier version of Lexical, but I haven't had the time to go back through each version to pinpoint which one introduced the bug, if any.

Lexical version: 0.8.0

## Steps To Reproduce

1. Go to the playground and type any word(s) (i.e. `any word(s)` :laughing: ).
2. Wrap any of these words with a preceding _accent grave_ character (`) and a succeeding one (a.k.a. the markdown shortcut for formating text as code). This can also be observed if you type three _accent grave_ characters in a row (a.k.a. the markdown shortcut for a code block).

## The current behavior

There's a leftover _accent grave_ character after the transformation.

## The expected behavior

There's no leftover _accent grave_ characters after the transformation.

**Comment by moy2010** [comment]
Just to be clear, by "ahead-of-time" I mean that the expected behaviour of an _accent grave_ character is to become the diacritic of a composite unicode character (i.e. à), which means that the usual behaviour of text editors with such markdown transformations usually only register the character after the key for it is pressed twice for each position.

I recall this being the behaviour in Lexical as well but, as mentioned above, I started observing the new behaviour quite recently.

**Comment by thegreatercurve** [comment]
I don't think we've changed any of the Markdown logic recently, but this works as expected for myself. I'm using a Mac, so if I type `OPTION` + `~` then I see the accent grave character. I then type "a" and it gets transformed to "à".

**Comment by moy2010** [comment]
But this wasn't about being able to type composite characters, @thegreatercurve  >.<. It's about the markdown shortcuts having an accent grave character as a leftover after the transformation for either the code format shortcut or the code block shortcut.

**Comment by moy2010** [comment]
That's really weird indeed :S. I tested it in the playground with both Chrome and Firefox before redacting the bug :(.

**Comment by moy2010** [comment]
Here's a recorded video:

https://watch.screencastify.com/v/O90sHrbX5LaOEZpziNyx

**Comment by moy2010** [comment]
I have investigated a bit further, and this is what is known as a [dead key](https://en.wikipedia.org/wiki/Dead_key).

I have tried to modify the regex to only match the unicode code point that represents the full character (U+0060), but it didn't work.

So far, the only work around is to modify my keyboard layout and set it to `no dead keys`, which is less than ideal... but it's something. 

Maybe something could be done to prevent the transformation from happening if it's a composition key, @trueadm ?
